WPThemeReview/Sniffs/Privacy/NoUrlShortenersSniff.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Sniffs\Privacy;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Check if the file contains a URL shortener.
+ *
+ * @since 0.2.0
+ */
+class NoUrlShortenersSniff implements Sniff {
+
+	/**
+	 * Error message template.
+	 *
+	 * @var string
+	 */
+	const ERROR_MSG = 'No URL shorteners should used in the theme. Found: "%s".';
+
+	/**
+	 * Found used shortener in a file
+	 *
+	 * @var string
+	 */
+	protected $shortener;
+
+	/**
+	 * List of url shorteners.
+	 *
+	 * @since 0.2.0
+	 *
+	 * @var array
+	 */
+	protected $url_shorteners = [
+		'goo.gl',
+		'bitly.com',
+		'polrproject.org',
+		'www.rebrandly.com',
+		'tinyurl.com',
+		'hootsuite.com/pages/owl',
+		'is.gd',
+		'buffer.com',
+		'df.ly',
+		'bit.do',
+	];
+
+	/**
+	 * Supported Tokenizers
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+		'JS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
```

Comment:
I see a lot of tokens here which are irrelevant.

My guess is that all you need is `Tokens::$textStringTokens` + `T_COMMENT` + `T_DOC_COMMENT_STRING`.
Depending on how on comments are tokenized in CSS and JS, maybe `T_DOC_COMMENT`, but that last one would need to be checked.

Reply:
Initially I thought about merging `Tokens::$textStringTokens` and `Tokens::$commentTokens` but in the `$textStringTokens`I have `HEREDOC` and I wasn't sure I needed that, and in the `$commentTokens` there were `PHPCS` tokens so I wanted to exclude those. I'll try with what you suggested and also add a test for those cases 👍